### PR TITLE
util: check empty StringSlice.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: pkg/typeutil/string_slice.go [hunk 1]
```diff
@@ -34,6 +34,10 @@ func (s *StringSlice) UnmarshalJSON(text []byte) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if len(data) == 0 {
+		*s = nil
```
Comment (siddontang) [inline pkg/typeutil/string_slice.go]: is it safe to use nil here? maybe empty slice?

Comment (nolouch) [inline pkg/typeutil/string_slice.go]: maybe directly return nil?

Comment (siddontang) [inline pkg/typeutil/string_slice.go]: no, if the StringSlice must be changed if it has items before.

Comment (disksing) [inline pkg/typeutil/string_slice.go]: It should be safe.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
Can we check empty slice in marshal function? if empty, no need to quote?

**Comment by disksing** [comment]
In fact, I'm not sure why it need to quote.. but marshal/unmarshal should match. cc @nolouch

**Comment by nolouch** [comment]
JSON format string values must be enclosed in double quotes.

File: pkg/typeutil/string_slice.go (hunk 1 above)
Comment (siddontang) [inline pkg/typeutil/string_slice.go]: so if a slice has data before but be unmarshaled with an empty data, the slice may become nil, is it safe?

Comment (disksing) [inline pkg/typeutil/string_slice.go]: I can't think of a reason that makes it unsafe.

**Pull request merged**